spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/MultipleRetryPolicy.scala

```diff
@@ -46,7 +52,17 @@ class MultipleRetryPolicy(maxRetryCount: Int)
       ex: DriverException,
       nbRetry: Int): RetryDecision = {
 
-    RetryDecision.rethrow()
+    ex match {
+      case _: NoHostAvailableException |
```

Comment:
Other than `OperationTimedoutException`, `OverloadedException`, `ReadFailureException`, and `WriteFailureException` I don't believe the other exceptions are evaluated in `RetryPolicy` since they are automatically tied on next host (except for `NoHostAvailableException` which just means there are no more hosts to try)

Another issue is that without knowing why a request failed (`ReadFailure`/`WriteFailure`), it's hard to know whether or not to retry.  My experience is that if you get a failure is that it's usually just going to fail again (i.e. tombstone threshold on reads, batch size failures on writes), but hard to know for sure without looking at the reasons.